src/EditorFeatures/Core/Tagging/AbstractAsynchronousTaggerProvider.TagSource_ProduceTags.cs

```diff
@@ -275,9 +349,16 @@ private async Task<VoidResult> RecomputeTagsAsync(bool highPriority, Cancellatio
                 // Once we've computed tags, pause ourselves if we're no longer visible.  That way we don't consume any
                 // machine resources that the user won't even notice.
                 PauseIfNotVisible();
-            }
 
-            return default;
+                // If we were computing with frozen partial semantics here, enqueue work to compute *without* frozen
+                // partial snapshots so we move to accurate results shortly. Note: when the queue goes to process this
+                // message, if it sees any other events asking for frozen-partial-semantics, it will process in that
+                // mode again, kicking the can down the road to finally end with non-frozen-partial computation
+                if (frozenPartialSemantics)
```

Comment:
❔ Should we also condition this on `Workspace.PartialSemanticsEnabled`? It seems hard to tell because the workspace where this work is queued (`RoslynVisualStudioWorkspace`) might be different from the workspace where the actual operation runs (`RemoteWorkspace`).

Reply:
realisitically, i don't think that will ever be relevant.  EditorFeatures is always running on VS for the forseeable future.  If that ever changes, it will be something that we'll have to review a lot of code substantively to ensure things are in good shape.  And we'd really only be doing that if we not only were adding another host, but also in that other host we didn't want to support frozen-partial.  All that is far too speculative imo :)